crates/ruma-client-api/src/profile/get_profile_field.rs

```diff
@@ -0,0 +1,329 @@
+//! `GET /_matrix/client/*/profile/{userId}/{key_name}`
+//!
+//! Get a field in the profile of the user.
+
+pub mod v3 {
+    //! `/v3/` ([spec])
+    //!
+    //! [spec]: https://github.com/matrix-org/matrix-spec-proposals/pull/4133
+
+    use std::marker::PhantomData;
+
+    use ruma_common::{api::Metadata, metadata, OwnedUserId};
+
+    use crate::profile::{
+        AnyProfileField, ProfileField, StaticProfileField, StaticProfileFieldVisitor,
+    };
+
+    const METADATA: Metadata = metadata! {
+        method: GET,
+        rate_limited: false,
+        authentication: None,
+        history: {
+            unstable("uk.tcpip.msc4133") => "/_matrix/client/unstable/uk.tcpip.msc4133/profile/{user_id}/{field}",
+            // 1.15 => "/_matrix/client/v3/profile/{user_id}/{field}",
+        }
+    };
+
+    /// Request type for the `get_profile_field` endpoint.
+    #[derive(Debug, Clone)]
+    #[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+    pub struct Request<F: ProfileField> {
+        /// The user whose profile will be fetched.
+        pub user_id: OwnedUserId,
+
+        /// The profile field to get.
+        pub field: F,
+    }
+
+    impl<F: ProfileField> Request<F> {
+        /// Creates a new `Request` with the given user ID and field.
+        pub fn new(user_id: OwnedUserId, field: F) -> Self {
```

Comment:
It's a bit ugly for types that implement `StaticProfileField` that we need to construct it when we don't need it.

I am not sure what the alternative is? Having separate `Request` and `Response` types for `StaticProfileField` and `AnyProfileField`?

Reply:
I changed to that API.